src/main/java/com/coveros/selenified/utilities/ReportOverview.java

```diff
@@ -259,15 +260,55 @@ private void writeTestResult(String status, String cssClass, String className, I
                 link += " " + LINK_START + getReportDir(iTestResult) + "/" + pdfFilename + LINK_MIDDLE + "PDF Report" + LINK_END;
             }
         }
+        String failure = "";
+        if (!"Pass".equals(status) && iTestResult.getThrowable() != null) {
+            String messageId = TestCase.getRandomString(10);
+            String message = getErrorMessage(iTestResult.getThrowable());
+            failure = ": <a data-toggle='collapse' href='#" + messageId + "' role='button' aria-expanded='false' aria-controls='" + messageId + "'>" + message + "</a>";
+            failure += "<div style='padding-left:20px; font-size:small;' class='collapse' id='" + messageId + "'>" + StringUtils.join(iTestResult.getThrowable().getStackTrace(), "<br/>") + "</div>";
+        }
         writer.print("<tr class='" + cssClass + "'>");
-        cell(browser.getDetails());
+        cell(getBrowser(browser));
         cell(Utils.escapeHtml(className));
         cell(Utils.escapeHtml(Reporter.capitalizeFirstLetters(iTestResult.getName())));
-        cell(status);
+        cell(status + failure);
         cell(link);
         writer.println(TR);
     }
 
+    /**
+     * Retrieves the high level error message from the thrown exception. If message exists, use that,
+     * if not, and localized message exists, use that, if not, but cause exists, use that, if not,
+     * returns "Unknown Error"
+     *
+     * @param threw the thrown exception
+     * @return String: the high level error message
+     */
+    private String getErrorMessage(Throwable threw) {
+        String errorMessage = "Unknown Error";
+        if (threw.getMessage() != null) {
+            errorMessage = threw.getMessage();
+        } else if (threw.getLocalizedMessage() != null) {
+            errorMessage = threw.getLocalizedMessage();
+        } else if (threw.getCause() != null) {
+            errorMessage = threw.getCause().toString();
+        }
+        return errorMessage;
+    }
+
+    /**
+     * Gets the browser details. If browser is null, returns unknown browser
+     *
+     * @param browser the browser the tests were run against
+     * @return String: the html friendly, human readable browser details
+     */
+    private String getBrowser(Browser browser) {
+        if (browser == null) {
```

Comment:
why not just let the null throw an NPE? It seems like if browser is null here then something has gone pretty wrong

Reply:
because then the report doesn't get generated at all